### PR TITLE
ERR: Improve error message when assigning a complete row using 'at' m…

### LINKED ISSUE #48729: BUG: DataFrame.at fails when setting a whole row
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame(index=['a'], columns=['col1', 'col2'])

df.at['a'] = ['123', '15']
print(df)
```

```python-traceback
Traceback (most recent call last):
  File "/home/*/.local/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 3800, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 144, in pandas._libs.index.IndexEngine.get_loc
TypeError: 'slice(None, None, None)' is an invalid key

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/*/pandas_bug.py", line 6, in <module>
    df.at['a'] = ['123', '15']
  File "/home/*/.local/lib/python3.10/site-packages/pandas/core/indexing.py", line 2438, in __setitem__
    return super().__setitem__(key, value)
  File "/home/*/.local/lib/python3.10/site-packages/pandas/core/indexing.py", line 2393, in __setitem__
    self.obj._set_value(*key, value=value, takeable=self._takeable)
  File "/home/*/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 4208, in _set_value
    icol = self.columns.get_loc(col)
  File "/home/*/.local/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 3807, in get_loc
    self._check_indexing_error(key)
  File "/home/*/.local/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 5963, in _check_indexing_error
    raise InvalidIndexError(key)
pandas.errors.InvalidIndexError: slice(None, None, None)
```

### Issue Description

Trying to set a row in a DataFrame using DataFrame.at fails on pandas 1.5.0. It worked on pandas 1.4.X.

Setting a single value (i.e. `df.at['a', 'col1'] = '15'`) works fine.
Setting the row with loc works as well (i.e. `df.loc['a'] = ['123', '15']`).

### Expected Behavior

The row values should be set:
```
  col1 col2
a  123   15
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 87cfe4e38bafe7300a6003a1d18bd80f3f77c763
python           : 3.10.4.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.10.16.3-microsoft-standard-WSL2
Version          : #1 SMP Fri Apr 2 22:23:49 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.0
numpy            : 1.23.1
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 59.6.0
pip              : 22.0.2
Cython           : 0.29.32
pytest           : 7.1.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.8.0
html5lib         : 1.1
pymysql          : None
psycopg2         : 2.9.3
jinja2           : 3.0.3
IPython          : None
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.1
snappy           : None
sqlalchemy       : 1.4.39
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by ambujpawar**
Better error reporting when trying to assign a new row using 'at' method

- [x] closes #48729 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/tests/indexing/test_at.py [hunk 1]
```diff
@@ -234,3 +234,10 @@ def test_at_categorical_integers(self):
             for key in [0, 1]:
                 with pytest.raises(KeyError, match=str(key)):
                     df.at[key, key]
+
+    def test_at_applied_for_rows(self):
+        # GH#48729 .at should raise InvalidIndexError when assigning rows
+        df = DataFrame(index=["a"], columns=["col1", "col2"])
+        new_row = [123, 15]
+        with pytest.raises(InvalidIndexError, match=str(new_row)):
```
Comment (mroeschke) [inline pandas/tests/indexing/test_at.py:242]: Could you `match` the entire message?

File: pandas/core/frame.py [hunk 2]
```diff
@@ -4239,6 +4240,14 @@ def _set_value(
                 self.loc[index, col] = value
             self._item_cache.pop(col, None)
 
+        except InvalidIndexError:
+            # GH48729: Seems like you are trying to assign a value to a
+            # row when only scalar options are permitted
+            raise InvalidIndexError(
```
Comment (mroeschke) [inline pandas/core/frame.py:4246]: Could you `raise InvalidIndexError(...) from` the original exception?

Comment (ambujpawar) [reply pandas/core/frame.py:4246]: The original exception is: 
`slice(None, None, None)
`
which I dont think is pretty useful as it is still does not let the user know what causes the issue

File: pandas/tests/indexing/test_at.py (hunk 1 above)
Comment (ambujpawar) [reply pandas/tests/indexing/test_at.py:242]: Done

File: pandas/core/frame.py (hunk 2 above)
Comment (ambujpawar) [reply pandas/core/frame.py:4246]: Would you really like to use that?

Comment (mroeschke) [reply pandas/core/frame.py:4246]: It's mainly to keep the context of where the original exception was raised by using `from original_err`

Comment (ambujpawar) [reply pandas/core/frame.py]: Hi, addressed the changes you requested.
Please let me know if this what you were referring to

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ambujpawar

**Comment by jorisvandenbossche** [comment]
Shall we backport this to 1.5.x? 
Re-reading https://github.com/pandas-dev/pandas/issues/48729, we started to raise the incomprehensible error in 1.5.0 (it actually worked before)

**Comment by mroeschke** [comment]
Sure, makes sense to backport

**Comment by mroeschke** [comment]
@Meeseeksdev backport 1.5.x

**Pull request merged**